core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClientHandlerFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A factory for creating a {@link ClientCircuitBreakerHandler}.
+ */
+@UnstableApi
+public interface CircuitBreakerClientHandlerFactory<CB, I extends Request>  {
```

Comment:
It seems we don't have to create the handler lazily using the factory.
Should we simply replace this factory with a static method in `CircuitBreakerClientHandler`?
For example:
```java
public interface CircuitBreakerClientHandler {
    static CircuitBreakerClientHandler of(CircuitBreaker cb) {
        return of((ctx, req) -> cb);
    }

    static CircuitBreakerClientHandler of(CircuitBreakerMapping cb) {
        return new DeafultCircuitBreakerClientHandler(cb);
    }
}

public interface Resilience4JClientCircuitBreakerHandler extends
    CircuitBreakerClientHandler {

    static CircuitBreakerClientHandler of(resilience4j.CircuitBreaker cb) {
        return of((ctx, req) -> cb);
    }

    static CircuitBreakerClientHandler of(resilience4j.CircuitBreakerMapping cb) {
        return new DeafultResilience4JCircuitBreakerClientHandler(cb);
    }
```

Reply:
I was wondering if it is possible to remove `Resilience4jCircuitBreakerClients` and let users use `Resilience4JClientCircuitBreakerHandler` to create a `CircuitBreakerClient`?

```java
CircuitBreakerClient.newDecorator(
    Resilience4JClientCircuitBreakerHandler.of(resilience4j.CircuitBreaker()));

CircuitBreakerClient.newDecorator(
    Resilience4JClientCircuitBreakerHandler.of(
        resilience4j.CircuitBreakerMapping.perMethod()));
```